console/src/view/task.rs

```diff
@@ -123,13 +163,134 @@ impl TaskView {
+        frame.render_widget(percentiles_widget_2, percentiles_columns[1]);
+        frame.render_widget(histogram_sparkline, sparkline_area);
+    }
+}
+
+fn make_chart_data(details: DetailsRef, task_id: u64, width: u16) -> (Vec<u64>, HistogramMetadata) {
+    details
+        .borrow()
+        .as_ref()
+        .and_then(|details| filter_same_task(task_id, details))
+        .and_then(|details| details.poll_times_histogram())
+        .map(|histogram| {
+            let step_size =
+                ((histogram.max() - histogram.min()) as f64 / width as f64).ceil() as u64 + 1;
+            let data = if step_size > 0 {
+                let mut found_first_nonzero = false;
+                let data: Vec<u64> = histogram
+                    .iter_linear(step_size)
+                    .map(|it| it.count_since_last_iteration())
+                    .filter(|count| {
+                        // Remove the 0s from the leading side of the buckets.
+                        // Because HdrHistogram can return empty buckets depending
+                        // on its internal state, as it approximates values.
+                        if *count == 0 && !found_first_nonzero {
+                            false
+                        } else {
+                            found_first_nonzero = true;
+                            true
+                        }
+                    })
+                    .collect();
+                data
+            } else {
+                Vec::new()
+            };
+            let max_bucket = data.iter().max().copied().unwrap_or_default();
+            let min_bucket = data.iter().min().copied().unwrap_or_default();
+            (
+                data,
+                HistogramMetadata {
+                    max_value: histogram.max(),
+                    min_value: histogram.min(),
+                    max_bucket,
+                    min_bucket,
+                },
+            )
+        })
+        .unwrap_or_default()
+}
+
+fn make_percentiles_widgets(details: DetailsRef, task_id: u64) -> (Text<'static>, Text<'static>) {
+    let percentiles_iter = details
+        .borrow()
+        .as_ref()
+        .and_then(|details| filter_same_task(task_id, details))
+        .and_then(|details| details.poll_times_histogram())
+        .map(|histogram| {
+            vec![
+                (5, histogram.value_at_percentile(5.0)),
+                (10, histogram.value_at_percentile(10.0)),
+                (25, histogram.value_at_percentile(25.0)),
+                (50, histogram.value_at_percentile(50.0)),
+                (75, histogram.value_at_percentile(75.0)),
+                (90, histogram.value_at_percentile(90.0)),
+                (95, histogram.value_at_percentile(95.0)),
+                (99, histogram.value_at_percentile(99.0)),
+            ]
+        })
+        .map(|pairs| {
+            pairs.into_iter().map(|pair| {
+                format!(
+                    "p{}: {:.prec$?}",
```

Comment:
nit: it would be nice to bold the percentile label ("pNN") like we do for labels elsewhere?

Reply:
Updated:
<img width="1440" alt="Screen Shot 2021-06-29 at 2 52 32 PM" src="https://user-images.githubusercontent.com/1333721/123852142-fdaec100-d8e9-11eb-80e0-f297cee222c9.png">

To add more percentiles with different lengths like p5 or p99.9 I need to account for the spacing better, but not sure if we care about those at the moment?

Reply:
alternatively, we could just make the poll times section a couple lines longer, and make one column of percentiles. then, the percentiles box wouldn't have to be 50% of the section, and we could make the histogram wider...and get more vertical space, which would (hopefully) allow us to draw the histogram with a little more vertical resolution.

but, we can tweak the layout and stuff in subsequent PRs. the important part for this branch, IMO, is getting the histogram wired through from the subscriber to the console, and having a basic setup for drawing it. which we've done! so, let's just merge this and keep working on making it look nice as a follow-up. :)